### PR TITLE
mac80211.sh shellcheck fixes

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh [hunk 1]
```diff
@@ -998,10 +998,10 @@ get_freq() {
 	iw "$phy" info | awk -v band="$band" -v channel="[$channel]" '
 
 $1 ~ /Band/ {
-	band_match = band == $2
+	band_match = band = $2
 }
```
Comment (jow-) [inline package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh:1002]: This is embedded Awk code, is this change really intended?

File: package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh [hunk 2]
```diff
@@ -998,10 +998,10 @@ get_freq() {
...
 
-band_match && $3 == "MHz" && $4 == channel {
+band_match && $3 = "MHz" && $4 = channel {
```
Comment (jow-) [inline package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh:1004]: This is embedded Awk code, is this change really intended?

File: package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh (hunk 1 above)
Comment (neheb) [reply package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh:1002]: Probably not.

Comment (neheb) [reply package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh]: Fixed.

File: package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh (hunk 2 above)
Comment (neheb) [reply package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh]: Fixed.

File: package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh [hunk 3]
```diff
@@ -646,7 +646,7 @@ find_phy() {
 		}
 	}
 	[ -n "$macaddr" ] && {
-		for phy in $(ls /sys/class/ieee80211 2>/dev/null); do
+		for phy in /sys/class/ieee80211/*; do
```
Comment (dangowrt) [inline package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh:649]: You should add
```
phy=${phy##*/}
```
to strip the path and only keep the phy name.
And you have to also cover the case of `/sys/class/ieee80211/` being empty or not even existing. In that case the loop content will be called once with phy="/sys/class/ieee80211/*"

Comment (neheb) [reply package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh:649]: so replace with find?

**Comment by neheb** [comment]
Fixed.

**Comment by neheb** [comment]
Rebased.

**Comment by neheb** [comment]
Forget it
